src/sinks/kafka/setup.rs

```diff
@@ -26,15 +27,30 @@ pub struct Config {
 
 impl SinkProvider for WithUtils<Config> {
     fn bootstrap(&self, input: StageReceiver) -> BootstrapResult {
-        let mut builder = Producer::from_hosts(self.inner.brokers.clone());
 
-        if let Some(timeout) = &self.inner.ack_timeout_secs {
-            builder = builder
-                .with_ack_timeout(Duration::from_secs(*timeout))
-                .with_required_acks(RequiredAcks::One)
+        let retry = || -> Result<Producer, String> {
+            for _ in 1..10 {
+                let mut builder = Producer::from_hosts(self.inner.brokers.clone());
+
+                if let Some(timeout) = &self.inner.ack_timeout_secs {
+                    builder = builder
+                        .with_ack_timeout(Duration::from_secs(*timeout))
+                        .with_required_acks(RequiredAcks::One)
+                };
+
+                match builder.create() {
+                    Ok(producer) => return Ok(producer),
+                    Err(err) => {
+                        println!("error while creating producer: {}", err);
+                        println!("waiting 3 secs...");
+                        sleep(Duration::from_secs(3))
+                    }
+                };
+            }
+            Err(String::from("giving up"))
         };
 
-        let producer = builder.create()?;
+        let producer = retry()?;
```

Comment:
Looks good overall! The closure idea is clean. I'd personally avoid the closure and just mutate producer directly:
```rs
let mut p = Error("giving up");
for ... {
    //...
    match ... {
        Ok(producer) => { p = Ok(producer); break }
        //...
    }
}
let producer = p?;
```
but that's a matter of style